week1/1.5/Notes/Notes/Controllers/Model Controllers/NoteController.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  NoteController.swift
+//  Notes
+//
+//  Created by Eric Lanza on 7/6/18.
+//  Copyright © 2018 ETLanza. All rights reserved.
+//
+
+import Foundation
+
+class NoteController: Codable {
+    
+    static let shared = NoteController()
+    
+    var notes = [Note]()
+    
+    //MARK: - CRUD Methods
+    func addNote(_ note: Note) {
+        notes.append(note)
+        saveToPersistentStore()
+    }
+    
+    func updateNote(_ note: Note, withBodyText bodyText: String) {
+        note.bodyText = bodyText
+        saveToPersistentStore()
+    }
+    
+    func deleteNote(_ note: Note) {
+        guard let indexOfNote = NoteController.shared.notes.index(of: note) else { return }
```

Comment:
As a general rule of thumb you can favor shorter variable names if they are created and immediately used. e.g. does the name `indexOfNote` save you from any thinking or searching as compared to the name `index` here? If the variable is going to propagate further through the code than you might ask "`index` of what?" but since you're using it immediately then it's not ambiguous.